src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1144,6 +1166,123 @@ export default class SourceFileLinter {
 		}
 	}
 
+	#analyzeNewAndApplySettings(node: ts.NewExpression | ts.CallExpression) {
+		node?.arguments?.filter((arg) => ts.isObjectLiteralExpression(arg))
+			.flatMap((arg) => arg.properties)
+			.forEach((prop) => {
+				if (ts.isPropertyAssignment(prop) &&
+					(ts.isCallExpression(node) ||
+						this.#isPropertyBinding(node, getPropertyNameText(prop.name)))
+				) {
+					if (ts.isObjectLiteralExpression(prop.initializer)) {
+						this.#analyzeControlBindings(prop.initializer, ["type"]);
+					} else {
+						this.#analyzeControlStringBindings(prop);
+					}
+				}
+			});
+	}
+
+	#analyzeControlBindings(node: ts.ObjectLiteralExpression, propNames: string[]) {
+		node?.properties.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop)) {
+				return;
+			}
+
+			// Get the type property
+			let typeField;
+			if (ts.isObjectLiteralExpression(prop.initializer)) {
+				typeField = propNames
+					.map((name) => getPropertyAssignmentInObjectLiteralExpression(
+						name, prop.initializer as ts.ObjectLiteralExpression))
```

Comment:
This casting is inevitable as TS does not detect properly that `prop.initializer` is `ObjectLiteralExpression` althought the if statement above

Reply:
This is because TypeScript can't be sure that between the `isObjectLiteralExpression` call and the usage within the map callback (different function context), the reference `prop.initializer` is still the same.

You can solve this by assigning `prop.initializer` to a local variable before the `.map` call, as in this case the reference can't change.

But maybe before changing that, please have a look at `getPropertyAssignmentsInObjectLiteralExpression` (plural form). Although I don't fully understand why the paramNames need to be passed from the outside, I think that would be the best approach to pick multiple properties from the given ObjectLiteralExpression.

Reply:
Done